plugin/metric/reporter/segment.go

```diff
@@ -0,0 +1,73 @@
+package reporter
+
+import (
+	"time"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/metric/collector"
+
+	"github.com/segmentio/analytics-go"
+	"go.uber.org/zap"
+)
+
+// Segment is the metrics collector https://segment.com/.
+type segment struct {
+	l           *zap.Logger
+	identifier  string
+	environment common.ReleaseMode
+	client      analytics.Client
+}
+
+const (
+	// identifyTraitForPlan is the trait key for subscription plan.
+	identifyTraitForPlan = "plan"
+	// metricValueField is the property key for value
+	metricValueField = "value"
+	// metricEnvironmentField is the property key for environment
+	metricEnvironmentField = "environment"
+)
+
+// NewSegmentReporter creates a new instance of segment
+func NewSegmentReporter(l *zap.Logger, key string, identifier string, mode common.ReleaseMode) MetricReporter {
+	client := analytics.New(key)
+
+	return &segment{
+		l:           l,
+		identifier:  identifier,
+		environment: mode,
```

Comment:
We should use different segment sources for different mode, but seems we also should keep the `environment` to be the `profile.mode` to avoid the hard code

Reply:
Updated, PTAL @tianzhou